src/Text-Core/RunArray.class.st

```diff
@@ -452,12 +419,10 @@ RunArray >> detectMin: aBlock [
 RunArray >> do: aBlock [
 	"This is refined for speed"
 
-	1 to: runs size do: [:i |
-		| r v |
-		v := values at: i.
-		r := runs at: i.
-		[( r := r - 1) >= 0]
-			whileTrue: [aBlock value: v]]
+	runs doWithIndex: [ :run :index |
```

Comment:
the initial implementation is faster since `to:do:` and `whileTrue:` are inlined.
Usually I do not care that much, but users of RunArray might need some kind of perf.

Reply:
I think that #timesRepeat: is now inlined too. @MarcusDenker do you know?